Parsers/ASimWebSession/Parsers/ASimWebSessionF5ASM.yaml

```diff
@@ -0,0 +1,169 @@
+  Link: https://aka.ms/ASimWebSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: F5 Big IP ASM Documentation
+  Link: 
+    https://techdocs.f5.com/en-us/bigip-15-0-0/external-monitoring-of-big-ip-systems-implementations/event-messages-and-attack-types.html
+    https://docplayer.net/9349757-Common-event-format-configuration-guide.html
+Description: |
+  This ASIM parser supports normalizing F5 Big IP ASM(Application Security Manager) logs to the ASIM Web Session normalized schema. F5 Big IP ASM events are captured through F5 Network data connector which ingests attack events and data into Microsoft Sentinel in CEF format over Syslog using F5's technology partner Arcsight.
+ParserName: ASimWebSessionF5ASM
+EquivalentBuiltInParser: _ASim_WebSession_F5ASM
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let DvcActionLookup = datatable (DeviceAction: string, DvcAction: string)
+  [
+      "Blocked", "Deny",
+      "blocked", "Deny",
+      "Passed", "Allow",
+      "passed", "Allow",
+      "Alerted", "Deny",
+      "alerted", "Deny"
+  ];
+  let EventSeverityLookup = datatable (LogSeverity: string, EventSeverity: string)
+  [
+      "0", "Low",
+      "1", "Low",
+      "2", "Low",
+      "3", "Low",
+      "4", "Medium",
+      "5", "Medium",
+      "6", "Medium",
+      "7", "High",
+      "8", "High",
+      "9", "High",
+      "10", "High"
+  ];
+  let parser=(disabled: bool=false) {
+      let AllData = CommonSecurityLog
+          | where DeviceVendor == "F5" and DeviceProduct == "ASM"
+          | invoke _ASIM_ResolveDvcFQDN('DeviceName')
+          | project-rename DvcIpAddr = DeviceAddress;
+      let GeneralEnforcementData = AllData
+          | where ((substring(DeviceEventClassID, 0, 1) == "2" and strlen(DeviceEventClassID) == 9) or DeviceEventClassID == Activity) and DeviceEventClassID !in ("Brute Force Attack", "IP Enforcer Attack", "Web Scraping Attack", "DoS Attack")
+          | parse-kv DeviceCustomString3 as (Host: string, ["User-Agent"]: string, Cookie: string, Referer: string) with (pair_delimiter="\\r\\n", kv_delimiter=":")
+          | parse DeviceCustomString3 with * "HTTP/" HttpVersion: string "\\r\\n" rest: string
+          | extend
+              EventResultDetails = tostring(FieldDeviceCustomNumber1)
+          | project-rename 
+              DstIpAddr = DestinationIP,
+              DstPortNumber = DestinationPort,
+              EventOriginalUid = ExtID,
+              HttpRequestMethod = RequestMethod,
+              NetworkApplicationProtocol = ApplicationProtocol,
+              HttpCookie = Cookie,
+              HttpHost = Host,
+              HttpReferrer = Referer,
+              HttpUserAgent = ['User-Agent'],
+              HttpRequestXff = DeviceCustomString5
+          | extend
+              EventResult = iff(toint(EventResultDetails) >= 400 or DeviceAction =~ "blocked", "Failure", "Success"),
+              HttpStatusCode = EventResultDetails,
```

Comment:
why no status code in failure?
![image](https://github.com/Azure/Azure-Sentinel/assets/97222872/409f81fd-7ceb-4c6b-a2f6-66a39aca8938)

Reply:
Status Code are available for GeneralEnforcement type of data only.
For Anoaly Detection type of data, we don't have status code value hence they are empty.